terminal.go

```diff
@@ -712,19 +712,28 @@ func (g *LightningTerminal) setUpLNDClients() error {
 	// because that doesn't do anything else than just connect. We'll check
 	// if lnd is also ready to be used in the next step.
 	log.Infof("Connecting basic lnd client")
-	err := wait.NoError(func() error {
-		// Create an lnd client now that we have the full configuration.
-		// We'll need a basic client and a full client because not all
-		// subservers have the same requirements.
-		var err error
+
+	for {
```

Comment:
should we have some kind of back off just? 
Cause at the moment, if something is actually wrong that needs user intervention, this will spin.

Reply:
also - if LiT is asked to shutdown during this loop - it should be able to exit from here which it currently wont.

Reply:
Ah yeah :)! Added checking of if we should shutdown before for every step in the loop.

 > should we have some kind of back off just?
 
 Wasn't exactly sure what this was referring to, but I've now added shutdown check + the user has the ability to end the execution with `CTRL+C`. The user also can see what's going wrong with the `litcli status` command, so I think that should be ok?

Reply:
> Wasn't exactly sure what this was referring to,

What I mean is that this could spin (ie go around the for loop very very fast  if something non-timeout related is erroring). So between iterations, we should add a bit of a wait.

This can be accomplished by moving your `checkRunning` to the end of the for loop and then replacing the default case with a timeout.

Reply:
Ah thanks for clarifying :)! I added a timeout for the duration of the `defaultStartupTimeout` to address this! Let me know if you think I should use some other timeout :)!